### ISSUE TITLE
GLOBAL IDENTITY FUNCTIONS

### ISSUE THREAD

**Issue by mshrestha12** [mshrestha12/TSQL#3]
**@@IDENTITY** 
It returns the last identity value generated for any table in the current session, across all scopes.
Let me explain this... suppose we create an insert trigger on table which inserts a row in another table with generate an identity column, then @@IDENTITY returns that identity record which is created by trigger.

**SCOPE_IDENTITY**
It returns the last identity value generated for any table in the current session and the current scope.
Let me explain this... suppose we create an insert trigger on table which inserts a row in another table with generate an identity column, then SCOPE_IDENTITY result is not affected but if a trigger or a user defined function is affected on the same table that produced the value returns that identity record then SCOPE_IDENTITY returns that identity record which is created by trigger or a user defined function.

**IDENT_CURRENT** 
It returns the last identity value generated for a specific table in any session and any scope.
In other words, we can say it is not affected by scope and session, it only depends on a particular table and returns that table related identity value which is generated in any session or scope.

**Comment by mshrestha12** [comment]
@@IDENTITY: **returns the last identity value generated on your SQL connection (SPID). Most of the time it will be what you want, but sometimes it isn't (like when a trigger is fired in response to an INSERT, and the trigger executes another INSERT statement).**

SCOPE_IDENTITY(): **returns the last identity value generated in the current scope (i.e. stored procedure, trigger, function, etc).**

IDENT_CURRENT(): returns the last identity value for a specific table. Don't use this to get the identity value from an INSERT, it's subject to race conditions (i.e. multiple connections inserting rows on the same table).

IDENTITY(): used when declaring a column in a table as an identity column.

https://msdn.microsoft.com/en-us/library/ms187342.aspx
